Common/src/Storage/TaskLifeCycleHelper.cs

```diff
@@ -0,0 +1,237 @@
+                                                                                    TaskStatus.Creating,
+                                                                                    request.Options ??
+                                                                                    throw new ArmoniKException("Task Options should not be null here"),
+                                                                                    new Output(false,
+                                                                                               ""))),
+                                cancellationToken)
+                   .ConfigureAwait(false);
+
+    await resultTable.SetTaskOwnership(sessionId,
+                                       taskCreationRequests.SelectMany(r => r.ExpectedOutputKeys.Select(key => (key, r.TaskId)))
+                                                           .AsICollection(),
+                                       cancellationToken)
+                     .ConfigureAwait(false);
+  }
+
+  public static async Task FinalizeTaskCreation(ITaskTable                       taskTable,
+                                                IResultTable                     resultTable,
+                                                IPushQueueStorage                pushQueueStorage,
+                                                ICollection<TaskCreationRequest> taskRequests,
+                                                string                           sessionId,
+                                                string                           parentTaskId,
+                                                ILogger                          logger,
+                                                CancellationToken                cancellationToken)
+  {
+    if (!taskRequests.Any())
+    {
+      return;
+    }
+
+    var parentExpectedOutputKeys = new List<string>();
+
+    if (!parentTaskId.Equals(sessionId))
+    {
+      parentExpectedOutputKeys.AddRange(await taskTable.GetTaskExpectedOutputKeys(parentTaskId,
+                                                                                  cancellationToken)
+                                                       .ConfigureAwait(false));
+    }
+
+    var taskDataModels = taskRequests.Select(request => new IResultTable.ChangeResultOwnershipRequest(parentExpectedOutputKeys.Intersect(request.ExpectedOutputKeys),
+                                                                                                      request.TaskId));
+    await resultTable.ChangeResultOwnership(sessionId,
+                                            parentTaskId,
+                                            taskDataModels,
+                                            cancellationToken)
+                     .ConfigureAwait(false);
+
+    var readyTasks = new List<(string taskId, int priority, string partitionId)>();
```

Comment:
Maybe `readyTasks` could be a `Dictionary<(string parittionId, int priority), string>`. This would provide the group by naturally during construction

Reply:
but (partitionid, priority) is not unique

Reply:
Sorry, I meant `List<string>`. Go to another PR.